--- WebKitTools/DumpRenderTree/LayoutTestController.h	(revision 39666)
+++ WebKitTools/DumpRenderTree/LayoutTestController.h	(working copy)
@@ -157,6 +157,7 @@ public:
     
     bool pauseAnimationAtTimeOnElementWithId(JSStringRef animationName, double time, JSStringRef elementId);
     bool pauseTransitionAtTimeOnElementWithId(JSStringRef propertyName, double time, JSStringRef elementId);
+    unsigned numberOfActiveAnimations() const;
     
 private:
     bool m_dumpAsText;
