--- trunk/scm/byeoru.scm	2007-11-09 23:19:10.000000000 +0900
+++ trunk/scm/byeoru.scm	2008-01-01 13:33:33.000000000 +0900
@@ -234,13 +234,34 @@
      ("n" (jungseong-u      . 3))
      ("m" (jungseong-eu     . 3))
      ;; Shifted keys
-     ("Q" (choseong-ssangbieub  . 5))
-     ("W" (choseong-ssangjieuj  . 5))
-     ("E" (choseong-ssangdigeud . 5))
-     ("R" (choseong-ssanggiyeog . 5) (jongseong-ssanggiyeog . 5))
-     ("T" (choseong-ssangsios   . 5) (jongseong-ssangsios   . 5))
-     ("O" (jungseong-yae        . 1))
-     ("P" (jungseong-ye         . 1)))))
+     ;; patched by nidev (nidev.plontra@gmail.com)
+     ;; http://nidev.org
+     ("Q" (choseong-ssangbieub   . 5))
+     ("W" (choseong-ssangjieuj   . 5))
+     ("E" (choseong-ssangdigeud  . 5))
+     ("R" (choseong-ssanggiyeog  . 5) (jongseong-ssanggiyeog  . 5))
+     ("T" (choseong-ssangsios    . 5) (jongseong-ssangsios    . 5))
+     ("Y" (jungseong-yo     . 1))
+     ("U" (jungseong-yeo    . 1))
+     ("I" (jungseong-ya     . 1))
+     ("O" (jungseong-yae     . 1))
+     ("P" (jungseong-ye      . 1))
+     ("A" (choseong-mieum   . 1) (jongseong-mieum   . (1 4)))
+     ("S" (choseong-nieun   . 1) (jongseong-nieun   . 3))
+     ("D" (choseong-ieung   . 1) (jongseong-ieung   . 1))
+     ("F" (choseong-rieul   . 1) (jongseong-rieul   . 3))
+     ("G" (choseong-hieuh   . 1) (jongseong-hieuh   . (1 4)))
+     ("H" (jungseong-o      . 3))
+     ("J" (jungseong-eo     . (1 4)))
+     ("K" (jungseong-a      . (1 4)))
+     ("L" (jungseong-i      . (1 4)))
+     ("Z" (choseong-kieuk   . 1) (jongseong-kieuk   . 1))
+     ("X" (choseong-tieut   . 1) (jongseong-tieut   . (1 4)))
+     ("C" (choseong-chieuch . 1) (jongseong-chieuch . 1))
+     ("V" (choseong-pieup   . 1) (jongseong-pieup   . (1 4)))
+     ("B" (jungseong-yu     . 1))
+     ("N" (jungseong-u      . 3))
+     ("M" (jungseong-eu     . 3)))))
 
 ;; The following definitions of 3-beol variants [final, 390, no-shift]
 ;; are based on the US keyboard layout.  A user of a different layout
