--- ecompiz-9999/plugins/switcher.c.orig	2009-02-04 15:05:42.000000000 +0900
+++ ecompiz-9999/plugins/switcher.c	2009-02-04 15:05:22.000000000 +0900
@@ -1468,7 +1468,7 @@
     }
 
   if (w->texture->pixmap &&
-      !(w->state & CompWindowStateHiddenMask)
+      !(w->state & CompWindowStateHiddenMask) && /* added by nidev */
       !(w->state & CompWindowStateShadedMask))
     {
       AddWindowGeometryProc oldAddWindowGeometry;
