--- configure.in.orig	2008-10-25 16:42:33.000000000 +0900
+++ configure.in	2008-10-25 16:44:08.000000000 +0900
@@ -50,7 +50,7 @@
 
 PKG_CHECK_MODULES(E, [enlightenment])
 
-datadir="${e_modules}/${PACKAGE}"    
+datadir=$(pkg-config --variable=modules enlightenment)/${PACKAGE}
 AC_ARG_ENABLE(homedir-install,
   AS_HELP_STRING([--enable-homedir-install], [Install module in homedir]),
   [ datadir="${HOME}/.e/e/modules/${PACKAGE}" ]
